cabal-install/Distribution/Client/Dependency.hs

```diff
@@ -451,6 +452,10 @@ standardInstallPolicy
   . hideInstalledPackagesSpecificBySourcePackageId
       [ packageId pkg | SpecificSourcePackage pkg <- pkgSpecifiers ]
 
+  . addDefaultSetupDependencies
+      (const [Dependency (PackageName "Cabal")
+              (orLaterVersion $ Version [1,24] [])])
+
```

Comment:
So make this function conditional on the build-type Custom, and then we don't need the previous patch to apply that conditionality within `addDefaultSetupDependencies` itself.